dsl_parser/functions.py

```diff
@@ -243,16 +285,20 @@ def __init__(self, args, **kwargs):
         super(GetProperty, self).__init__(args, **kwargs)
 
     def parse_args(self, args):
-        if not isinstance(args, list) or len(args) < 2:
+        if not _is_legal_nested_attribute_path(args):
             raise ValueError(
                 'Illegal arguments passed to {0} function. Expected: '
-                '<node_name, property_name [, nested-property-1, ... ]> but '
-                'got: {1}.'.format(self.name, args))
+                '[ node_name\\function, property_name\\function '
+                '[, nested-property-1\\function, ... ] '
+                'but got: {1}.'.format(self.name, args))
         self.node_name = args[0]
         self.property_path = args[1:]
 
     def validate(self, plan):
-        self.evaluate(plan)
+        # Try to evaluate only when the arguments don't contain functions
+        args = [self.node_name] + self.property_path
+        if all(map(lambda x: not isinstance(x, dict), args)):
```

Comment:
i think `if all(not isinstance(x, dict) for x in args)` is nicer, or maybe `if any(isinstance(x, dict) for x in args): return`

Reply:
ditto for all usages of map+lambda, it's not a big difference but in general i'd prefer generator expressions like that because 1) i think it's more readable (less cruft from `lambda`), 2) it's slightly faster (map does a whole function call which is slow in python and a genexp just executes the expression), 3) on python 2, map returns a list, so the `all()` won't be able to short-circuit (it will need to go over all `args` first, whereas the genexp would stop at the first false)

Reply:
Didn't know that (also forgot about generators)!